clang/include/clang/Basic/DiagnosticFrontendKinds.td

```diff
@@ -313,4 +313,11 @@ def warn_profile_data_unprofiled : Warning<
   InGroup<ProfileInstrUnprofiled>;
 } // end of instrumentation issue category
 
+def warn_sycl_wrong_aspect_usage : Warning<
+  "function '%0' uses aspect '%1' not listed in `sycl::device_has()`\n"
```

Comment:
> I've never seen diagnostics that come from LLVM pass, but emitted by clang. Interesting. Although I have concerns about how they will look like for the end users. I think we have some passes that emit errors, but they don't involve clang, so the quality of diagnostics is the same (no source info advantage), but the code is simpler. See #5706 for example. Maybe it will be better?

Yea I'm not familiar with this either. @maksimsab are there some other examples you can point us to? I'm not sure how to review FE changes here because I've never seen it done this way.

Reply:
Diagnostic message was refactored in f381619913908ff90da025bf1e50be4ef76acd26